var/spack/repos/builtin/packages/py-h5sh/package.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyH5sh(PythonPackage):
+    """Shell-like environment for HDF5."""
+
+    homepage = "https://h5sh.readthedocs.io/en/"
+    url      = "https://github.com/sethrj/h5sh/archive/v0.1.0.tar.gz"
+
+    maintainers = ['sethrj']
+
+    version('0.1.0', sha256='52d672e3140323e51084a1bfa2d84012e23c640469242381090a3c7af3e00e4f')
+
+    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
```

Comment:
I would remove this line as it isn't actually enforced by `python_requires`

Reply:
I'm confused, I thought that the requirements (including python version) had to be mirrored in spack rather than relied upon with the setup.py script. Has this changed? The `py-numpy` package for example contains explicit python requirements.

Reply:
They do, however, packages like numpy or matplotlib actually enforce this. For example, in the numpy `setup.py`:
```python
if sys.version_info[:2] < (3, 5):                                               
    raise RuntimeError("Python version >= 3.5 required.")  
```
In the matplotlib `setup.py`:
```python
        python_requires='>=3.6',                                                
```
These are the two main methods of enforcing Python version that I know of. The latter is specific to setuptools.

Since you're the developer of the package, I trust you to know which versions of Python are actually supported, so you can keep this line if you want. But for other packages, I never rely on the Python versions in classifiers as they are often versions that are "tested" but not necessarily all versions that would work.

Reply:
Gotcha. I will remove it since right now it realistically isn't restricting any python versions.